app/middleware/renderer.js

```diff
@@ -1,7 +1,51 @@
+const log = require('../lib/logger');
+const messages = require('../lib/messages');
+
 function findHelp(req, res) {
   res.render('find-help');
 }
 
+function location(req, res) {
+  if (res.locals.locationHeading === null) {
+    return res.render('error');
+  }
+  return res.render('location');
+}
+
+function results(req, res) {
+  res.render('results');
+}
+
+function emptyPostcodePage(req, res) {
+  log.debug('Empty Search');
+  res.locals.errorMessage = messages.emptyPostcodeMessage();
+  location(req, res);
+}
+
+function postcodeError(error, postcode, res, next) {
```

Comment:
For consistency with the other functions it would be good have the parameter list begin with `req, res, ...`.
Also, this function doesn't render a view within the function as it delegates that to the global error handling by passing an error into `next`. I suggest this function is removed and the logging and next call are moved back to current caller of this function. I don't setting `res.locals.errorMessage` does anything in this instance either as the error view has no use of that variable.